src/encoded/types/experiment.py

```diff
@@ -177,6 +176,102 @@ def related_series(self, request, related_series):
     def superseded_by(self, request, superseded_by):
         return paths_filtered_by_status(request, superseded_by)
 
+    @calculated_property(schema={
+        "title": "Protein tags",
+        "description": "The protein tags introduced through the genetic modifications of biosamples investigated in the experiment.",
+        "comment": "Do not submit. This field is calculated through applied_modifications.",
+        "type": "array",
+        "notSubmittable": True,
+        "items": {
+            "title": "Protein tag",
+            "description": "The protein tag introduced in the modification.",
+            "type": "object",
+            "additionalProperties": False,
+            "properties": {
+                "name": {
+                    "title": "Tag name",
+                    "type": "string",
+                    "enum": [
+                        "3xFLAG",
+                        "6XHis",
+                        "DsRed",
+                        "eGFP",
+                        "ER",
+                        "FLAG",
+                        "GFP",
+                        "HA",
+                        "mCherry",
+                        "T2A",
+                        "TagRFP",
+                        "TRE",
+                        "V5",
+                        "YFP",
+                        "mAID-mClover",
+                        "mAID-mClover-NeoR",
+                        "mAID-mClover-Hygro"
+                    ]
+                },
+                "location": {
+                    "title": "Tag location",
+                    "type": "string",
+                    "enum": [
+                        "C-terminal",
+                        "internal",
+                        "N-terminal",
+                        "other",
+                        "unknown"
+                    ]
+                },
+                "target": {
+                    "title": "Tagged protein",
+                    "type": "string",
+                    "linkTo": "Target",
+                }
+            }
+        }
+    })
+    def protein_tags(self, request, replicates=None):
+        protein_tags = []
+        protein_tags_lists = []
+        one_biosample_tags = []
+        if replicates is not None:
+            for rep in replicates:
+                replicateObject = request.embed(rep, '@@object?skip_calculated=true')
+                if replicateObject['status'] == 'deleted':
+                    continue
+                if 'library' in replicateObject:
+                    libraryObject = request.embed(replicateObject['library'], '@@object?skip_calculated=true')
+                    if libraryObject['status'] == 'deleted':
+                        continue
+                    if 'biosample' in libraryObject:
+                        biosampleObject = request.embed(libraryObject['biosample'], '@@object')
+                        if biosampleObject['status'] == 'deleted':
```

Comment:
Not sure if other status like "revoked" should be excluded here?

Reply:
Not sure. For calculated biosample summary, only "deleted" objects were excluded. But I guess it wouldn't hurt to also consider revoked?